sdk/servicebus/Azure.Messaging.ServiceBus/src/Diagnostics/ServiceBusEventSource.cs

```diff
@@ -441,7 +441,7 @@ public virtual void CompleteMessageException(string identifier, string exception
         }
 
         [Event(DeferMessageStartEvent, Level = EventLevel.Informational, Message = "{0}: DeferAsync start. MessageCount = {1}, LockToken = {2}")]
-        public virtual void DeferMessageStart(string identifier, int messageCount, string lockToken)
+        public virtual void DeferMessageStart(string identifier, int messageCount, Guid lockToken)
```

Comment:
This is unrelated to my change but a question for further improvements. The event source uses the `object[] args` overload to write events which causes value types to be boxed. I would actually recommend to use the `WriteEventCore` and stackalloc the struct. Example

https://github.com/dotnet/runtime/blob/01b7e73cd378145264a7cb7a09365b41ed42b240/src/libraries/System.Private.CoreLib/src/System/Threading/NativeRuntimeEventSource.PortableThreadPool.cs#L146-L153
or

https://github.com/dotnet/runtime/blob/01b7e73cd378145264a7cb7a09365b41ed42b240/src/libraries/System.Threading.Tasks.Parallel/src/System/Threading/Tasks/ParallelETWProvider.cs#L233-L263

It would require though to allow unsafe. Are you ok with that? As far as I'm aware there is no consequence on the consumer of the lib when allowing unsafe

Reply:
/cc @pakrym do you have any thoughts on this?

Reply:
I would rather wait for the eventsource code generator that dotnet should soon release. Hopefully, it would generate code with optimizations already applied.

Reply:
Why wait? Are you worried about implications of unsafe?

Reply:
The other thing we could try would be to reorder the parameters so we can call into this overload - https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.tracing.eventsource.writeevent?view=net-5.0#System_Diagnostics_Tracing_EventSource_WriteEvent_System_Int32_System_String_System_String_System_String_

Reply:
This is probably worthy of a separate issue.

One other thing that makes me somewhat nervous is using a Guid in the method signature for an event. I've seen things like this fail silently before - we should verify that we have coverage for this.

Reply:
There are event source tests that I had to touch too. Shouldn't they cover it?

Reply:
I'm not sure that they would because those are unit tests and the IsEnabled call would return false. 
I'm thinking something more along the lines of what is in EventSourceLiveTests.

Reply:
The other thing is that changing the call to WriteEvent could be considered a breaking change for strongly typed consumers of these events. It would probably be best to add wrapper NonEvents that take the Guid and then only call ToString if IsEnabled when calling the actual Events.

Reply:
OK will switch back and do that